crates/networking/rpc/engine/fork_choice.rs

```diff
@@ -58,36 +58,56 @@ impl RpcHandler for ForkChoiceUpdatedV3 {
     }
 
     fn handle(&self, context: RpcApiContext) -> Result<Value, RpcErr> {
-        let storage = &context.storage;
         info!(
             "New fork choice request with head: {}, safe: {}, finalized: {}.",
             self.fork_choice_state.head_block_hash,
             self.fork_choice_state.safe_block_hash,
             self.fork_choice_state.finalized_block_hash
         );
-        let fork_choice_error_to_response = |error| {
-            let response = match error {
-                InvalidForkChoice::NewHeadAlreadyCanonical => ForkChoiceResponse::from(
-                    PayloadStatus::valid_with_hash(latest_canonical_block_hash(storage).unwrap()),
-                ),
-                InvalidForkChoice::Syncing => ForkChoiceResponse::from(PayloadStatus::syncing()),
-                reason => {
-                    warn!("Invalid fork choice state. Reason: {:#?}", reason);
-                    return Err(RpcErr::InvalidForkChoiceState(reason.to_string()));
-                }
-            };
-
-            serde_json::to_value(response).map_err(|error| RpcErr::Internal(error.to_string()))
-        };
 
         let head_block = match apply_fork_choice(
-            storage,
+            &context.storage,
             self.fork_choice_state.head_block_hash,
             self.fork_choice_state.safe_block_hash,
             self.fork_choice_state.finalized_block_hash,
         ) {
             Ok(head) => head,
-            Err(error) => return fork_choice_error_to_response(error),
+            Err(error) => {
+                let fork_choice_response = match error {
+                    InvalidForkChoice::NewHeadAlreadyCanonical => {
+                        ForkChoiceResponse::from(PayloadStatus::valid_with_hash(
+                            latest_canonical_block_hash(&context.storage).unwrap(),
+                        ))
+                    }
+                    InvalidForkChoice::Syncing => {
+                        // Start sync
+                        let current_number = context.storage.get_latest_block_number()?.unwrap();
```

Comment:
Can we avoid this unwrap?

Reply:
There is an issue for this #791,  the method should never return an option, so I replaced it with an unwrap or default instead